src/i18n-support/src/main/java/engineering/everest/lhotse/i18n/TranslatingValidator.java

```diff
@@ -0,0 +1,20 @@
+package engineering.everest.lhotse.i18n;
+
+import engineering.everest.lhotse.i18n.exceptions.TranslatableIllegalArgumentException;
+import engineering.everest.lhotse.i18n.exceptions.TranslatableIllegalStateException;
+
+@SuppressWarnings("PMD.ClassNamingConventions")
+public class TranslatingValidator {
+
+    public static void isTrue(boolean expression, String i18nMessageKey, Object... args) {
+        if (!expression) {
+            throw new TranslatableIllegalArgumentException(i18nMessageKey, args);
+        }
+    }
+
+    public static void isValidState(boolean expression, String i18nMessageKey, Object... args) {
+        if (!expression) {
+            throw new TranslatableIllegalStateException(i18nMessageKey, args);
+        }
+    }
+}
```

Comment:
I got her by trying to follow the pattern in the Apache Commons validator. I will remove it since it doesn't really add anything apart from confusion.

Reply:
I got "her" 💯

Reply:
Ha! She was charmed by my copy-and-paste programming style.

Reply:
Have a look at the changes just pushed